test/e2e/backup-resource/deletion.go

```diff
@@ -0,0 +1,165 @@
+
+// Test backup and restore of Kibishi using restic
+
+func Backup_deletion_with_snapshots() {
+	backup_deletion_test(true)
+}
+
+func Backup_deletion_with_restic() {
+	backup_deletion_test(false)
+}
+func backup_deletion_test(useVolumeSnapshots bool) {
+	var (
+		backupName string
+	)
+
+	client, err := NewTestClient()
+	Expect(err).To(Succeed(), "Failed to instantiate cluster client for backup tests")
+
+	BeforeEach(func() {
+		if useVolumeSnapshots && VeleroCfg.CloudProvider == "kind" {
+			Skip("Volume snapshots not supported on kind")
+		}
+		var err error
+		flag.Parse()
+		UUIDgen, err = uuid.NewRandom()
+		Expect(err).To(Succeed())
+		if VeleroCfg.InstallVelero {
+			Expect(VeleroInstall(context.Background(), &VeleroCfg, "", useVolumeSnapshots)).To(Succeed())
+		}
+	})
+
+	AfterEach(func() {
+		if VeleroCfg.InstallVelero {
+			err = VeleroUninstall(context.Background(), VeleroCfg.VeleroCLI, VeleroCfg.VeleroNamespace)
+			Expect(err).To(Succeed())
+		}
+	})
+
+	When("kibishii is the sample workload", func() {
+		It("should be successfully backed up and restored to the default BackupStorageLocation", func() {
+			backupName = "backup-" + UUIDgen.String()
+			// Even though we are using Velero's CloudProvider plugin for object storage, the kubernetes cluster is running on
+			// KinD. So use the kind installation for Kibishii.
+
+			Expect(runBackupDeletionTests(client, VeleroCfg.VeleroCLI, VeleroCfg.CloudProvider, VeleroCfg.VeleroNamespace, backupName, "", useVolumeSnapshots, VeleroCfg.RegistryCredentialFile, "us-west-1", VeleroCfg.BSLPrefix, VeleroCfg.BSLConfig)).To(Succeed(),
+				"Failed to successfully backup and restore Kibishii namespace")
+		})
+	})
+}
+
+// runUpgradeTests runs upgrade test on the provider by kibishii.
+func runBackupDeletionTests(client TestClient, veleroCLI, providerName, veleroNamespace, backupName, backupLocation string,
+	useVolumeSnapshots bool, registryCredentialFile, region, bslPrefix, bslConfig string) error {
+
+	oneHourTimeout, _ := context.WithTimeout(context.Background(), time.Minute*60)
+
+	if err := CreateNamespace(oneHourTimeout, client, deletionTest); err != nil {
+		return errors.Wrapf(err, "Failed to create namespace %s to install Kibishii workload", deletionTest)
+	}
+	defer func() {
+		if err := DeleteNamespace(context.Background(), client, deletionTest, true); err != nil {
+			fmt.Println(errors.Wrapf(err, "failed to delete the namespace %q", deletionTest))
+		}
+	}()
+
+	if err := KibishiiPrepareBeforeBackup(oneHourTimeout, client, providerName, deletionTest, registryCredentialFile); err != nil {
+		return errors.Wrapf(err, "Failed to install and prepare data for kibishii %s", deletionTest)
+	}
+	err := ObjectsShouldNotBeInBucket(VeleroCfg.CloudProvider, VeleroCfg.CloudCredentialsFile, region, VeleroCfg.BSLBucket, VeleroCfg.BSLPrefix, VeleroCfg.BSLConfig, backupName, BackupObjectsPrefix, 1)
+	if err != nil {
+		fmt.Println(errors.Wrapf(err, "Failed to get object from bucket %q", backupName))
```

Comment:
Maybe there is no need to print errors info, the same errors will be output again outside of caller function